src-input/duk_bi_math.c

```diff
@@ -90,6 +90,54 @@ DUK_LOCAL double duk__fmax_fixed(double x, double y) {
 	return duk_double_fmax(x, y);
 }
 
+DUK_LOCAL double duk__cbrt_fixed(double x) {
+	/* cbrt() is C99.  To avoid hassling embedders with the need to provide a
+	 * cube root function, we can get by with pow().  The result is not
+	 * identical, but that's OK: ES6 says it's implementation-dependent.
+	 */
+
+	duk_small_int_t c = (duk_small_int_t) DUK_FPCLASSIFY(x);
+
+#if defined(DUK_CBRT)
+	/* cbrt() matches ES6 requirements. */
+	return DUK_CBRT(x);
+#else
+	/* pow() does not, however. */
+	if (c == DUK_FP_NAN || c == DUK_FP_INFINITE || c == DUK_FP_ZERO) {
+		return x;
+	}
+	if (DUK_SIGNBIT(x)) {
+		return -DUK_POW(-x, 1.0 / 3.0);
+	} else {
+		return DUK_POW(x, 1.0 / 3.0);
+	}
+#endif
+}
+
+DUK_LOCAL double duk__log2_fixed(double x) {
+#if defined(DUK_LOG2)
+	return DUK_LOG2(x);
+#else
+	return DUK_LOG(x) / DUK_LOG(2.0);
```

Comment:
That's quite likely, but they can't do so when `log()` is provided by a custom library which the compiler doesn't know about. But I can make the change separately.